### PR TITLE
OBPIH-5809 Approval rollback - Approvals end to end workflow (fix after QA)

### PR DESCRIPTION
**Pull request by alannadolny**

### CODE REVIEW

**Review by awalkowiak — APPROVED** [review]

**Review by drodzewicz — APPROVED** [review]

File: grails-app/views/stockMovement/show.gsp [hunk 1]
```diff
@@ -90,7 +90,11 @@
             <g:isUserInAllRoles location="${stockMovement?.origin?.id}" roles="${[RoleType.ROLE_REQUISITION_APPROVER]}">
                 <g:set var="userHasRequestApproverRole" value="${true}"/>
             </g:isUserInAllRoles>
-            <g:set var="isApprovalRequired" value="${stockMovement?.requisition?.approvalRequired && stockMovement?.origin?.isApprovalRequired()}" />
+            <g:set var="isApprovalRequired"
```
Comment (awalkowiak) [inline grails-app/views/stockMovement/show.gsp:93]: This also could be moved to the `StockMovement`

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: grails-app/views/stockMovement/show.gsp [hunk 2]
```diff
@@ -90,7 +90,11 @@
...
+                   value="${stockMovement?.requisition?.approvalRequired &&
+                           stockMovement?.origin?.isApprovalRequired() &&
+                           stockMovement.status < RequisitionStatus.PICKING}"
```
Comment (kchelstowski) [inline grails-app/views/stockMovement/show.gsp:96]: I feel like this is dangerous - won't it compare the statuses chronologically?

File: grails-app/views/stockMovement/show.gsp [hunk 3]
```diff
@@ -90,7 +90,11 @@
             <g:isUserInAllRoles location="${stockMovement?.origin?.id}" roles="${[RoleType.ROLE_REQUISITION_APPROVER]}">
                 <g:set var="userHasRequestApproverRole" value="${true}"/>
             </g:isUserInAllRoles>
-            <g:set var="isApprovalRequired" value="${stockMovement?.requisition?.approvalRequired && stockMovement?.origin?.isApprovalRequired()}" />
+            <g:set var="isApprovalRequired"
+                   value="${stockMovement?.requisition?.approvalRequired &&
+                           stockMovement?.origin?.isApprovalRequired() &&
```
Comment (kchelstowski) [inline grails-app/views/stockMovement/show.gsp:95]:
```suggestion
                           stockMovement?.origin?.approvalRequired &&
```

File: grails-app/views/stockMovement/show.gsp (hunk 2 above)
Comment (kchelstowski) [reply grails-app/views/stockMovement/show.gsp:96]: Are you sure? it's `compare` method which is not the same as `compareTo` method available in `Comparator` interface. Those are not the same.

Comment (awalkowiak) [reply grails-app/views/stockMovement/show.gsp]: @alannadolny yup please make sure about that one

Comment (alannadolny) [reply grails-app/views/stockMovement/show.gsp]: This topic was previously answered by Justin: https://github.com/openboxes/openboxes/pull/4133#discussion_r1255253493

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**